### ISSUE TITLE
feature: analyze pull requests when merged

### ISSUE THREAD

**Issue by raix** [semantic-release/semantic-release#536]
Given I have a pull request ready to be merged and it's missing the angular commit format
when I add `feat: foo` to the merge message and I merge the pull request then nothing gets released.

Would it be possible to have semantic release look at the merge message body for angular commit messages?

My current workaround is to do a "dummy" commit triggering the release on CI 😐

**Comment by gr2m** [comment]
That is a great use case for github apps :) It’s fairly simple to do, e.g. check out https://probot.github.io/apps/validate-commit-msg

**Comment by pvdlg** [comment]
Yep. But for info this app uses [validate-commit-msg](https://github.com/conventional-changelog-archived-repos/validate-commit-msg) which is deprecated....

**Comment by pvdlg** [comment]
Feedback on this subject welcomed in #585

**Comment by myii** [comment]
Our method for dealing with this using Travis CI is to have [3 stages defined](https://github.com/saltstack-formulas/template-formula/blob/c71915ed1e38d72c38360d409ddcffd01a61f557/.travis.yml#L1-L5), where the `release` stage only runs when merging the PR.  [`commitlint`](https://github.com/saltstack-formulas/template-formula/blob/c71915ed1e38d72c38360d409ddcffd01a61f557/.travis.yml#L35-L43) runs for PRs as well, so that we can ensure that the commit messages will pass through to the `release` stage appropriately.

**Comment by travi** [comment]
I've found [this bot](https://github.com/probot/semantic-pull-requests) very helpful when using the default convention.

**Comment by myii** [comment]
@travi Thanks for that link.  I did find another bot previously but it is no longer a GitHub app.  Can't recall its name right now.
